File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.java
class Solution {
    public int findGCD(int[] nums) {
        int min = nums[0], max = nums[0];
        
        for(int i = 1; i < nums.length; i++){
            if(nums[i] > max)
                max = nums[i];
            if(nums[i] < min)
                min = nums[i];
        }
        int res = min_max_gcd(min,max);
        return res;
    }
    
    public int min_max_gcd(int min, int max) {
        if (min == max)
            return min;
        for(int i = max/2; i >= 1; i--){
            if(max % i == 0 && min % i == 0)
                return i;
        }
        return 1;
    }
}

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.swift
class Solution {
    func findGCD(_ nums: [Int]) -> Int {
        var min = 1001
        var max = 0
        
        for i in nums {
            if i > max { max = i }
            if i < min { min = i }
        }
        
        var x = 0
        while min != 0 {
            x = max
            max = min
            min = x % max
        }
        return max
    }
}